app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,52 @@
+class QuestionsController < ApplicationController
+  before_action :ensure_logged_in, only:[:new, :create, :update]
```

Comment:
I'd be tempted to move this to ApplicationController and use skip_before_action in the (few) controllers you don't want it to apply